### ISSUE TITLE
Support for sub-jets

### ISSUE THREAD

**Issue by jlingema** [HEP-FCC/fcc-edm#45]
Need some support to save sub-jet structures that go beyond float values (i.e. actual sub-jets). Something along the lines of

```yaml
  fcc::XYZJet :
    OneToManyRelations :
     - fcc::Jet subjets // Result of the tagging algorithm
    OneToOneRelations :
     - fcc::Jet jet // Reference to the jet
```

**Comment by cbernet** [comment]
Or shouldn't it be 

```yaml
  fcc::XYZJet :
    OneToManyRelations :
     - fcc::XYZJet subjets // Result of the tagging algorithm
    OneToOneRelations :
     - fcc::XYZJet jet // Reference to the jet
```

Where XYZJet would be GenJet or Jet, for example? 

Edit: 

no, that's stupid. Basically, a GenJet should have GenJets as subjets. 
So maybe we need: 

```yaml
  fcc::ResolvedGenJet :
    OneToManyRelations :
     - fcc::GenJet subjets // Result of the tagging algorithm
    OneToOneRelations :
     - fcc::GenJet jet // Reference to the jet
```

and so on.

**Comment by jlingema** [comment]
Addressed in #46
